syntax = "proto3";

service CacheService {
    rpc RegisterNode(NodeInfo) returns (Response);
    rpc DeregisterNode(NodeInfo) returns (Response);
    rpc Get(Key) returns (CacheItem);
    rpc Put(CacheItem) returns (Response);
    rpc Remove(Key) returns (Response);
}

message Key {
    string key = 1;
}

message CacheItem {
    string key = 1;
    string value = 2;
}

message NodeInfo {
    string ip = 1;
    int32 port = 2;
}

message Response {
    bool success = 1;
    string message = 2;
}
